### PR TITLE
consider distance field padding for spheres

### PR DESCRIPTION
**Pull request by v4hn**
Addresses a rather small inconsistency I pointed out [here](https://github.com/ros-planning/moveit/pull/3504#pullrequestreview-1657191555).

Two more helper patches are included to make the test nicer and the API slightly better.

### CODE REVIEW

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 1]
```diff
@@ -52,8 +52,9 @@ collision_detection::determineCollisionSpheres(const bodies::Body* body, Eigen::
     const auto radius = body->getDimensions()[0];
```
Comment (rhaschke) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp:52]: This radius doesn't consider padding and scaling. I guess that's what you wanted to consider?

File: moveit_core/utils/src/robot_model_test_utils.cpp [hunk 2]
```diff
@@ -265,6 +265,17 @@ RobotModelBuilder& RobotModelBuilder::addVisualBox(const std::string& link_name,
   return *this;
 }
 
+RobotModelBuilder& RobotModelBuilder::addCollisionSphere(const std::string& link_name, double radius,
+                                                         geometry_msgs::Pose origin)
+{
+  urdf::CollisionSharedPtr coll(new urdf::Collision);
+  urdf::SphereSharedPtr geometry(new urdf::Sphere);
```
Comment (rhaschke) [inline moveit_core/utils/src/robot_model_test_utils.cpp:272]: This should use std::make_shared

**Review by rhaschke — APPROVED** [review]
I will push fixes for the suggested changes.

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp (hunk 1 above)
Comment (v4hn) [reply moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]: No need for scaling when it checks only for 0.0, but fair enough! :)

**Review by rhaschke — APPROVED** [review]

**Pull request merged**